### ISSUE TITLE
Bug: wrap with "maxwidth" prefers wrapping before dot-accessors rather than wrapping arguments/parameters & after assignment operator

### ISSUE THREAD

**Issue by 1oo7** [nicklockwood/SwiftFormat#615]
Bug: wrap with "maxwidth" prefers wrapping before dot-accessors rather than wrapping arguments/parameters.

## How to Reproduce

Given the following line:

```
            if let actionModel = actionModel, let routerContainer = AsdfQwerSubActionRouterContainerMapping.routerContainerFor(actionModel: actionModel, asdfModel: asdfModel) {
```

We expect a `--maxwidth 80` with wrap first settings across all options to result in:

```
            if let actionModel = actionModel, let routerContainer =     
                AsdfQwerSubActionRouterContainerMapping.routerContainerFor(
                    actionModel: actionModel, 
                    asdfModel: asdfModel
                ) 
            {
```

However instead SwiftFormat wraps like this:

```
            if let actionModel = actionModel,
                let routerContainer = BenefitsSubActionRouterContainerMapping
                .routerContainerFor(actionModel: actionModel, planModel: planModel) {
```

## Thoughts

I feel that line breaks should not prefer to wrap just before a dot accessor over wrapping the arguments/parameters. There should at least be a way to disable this behavior. 

In Apple's `swift-format` there is a configuration option, lineBreakAroundMultilineExpressionChainComponents, which you can set to false to avoid this behavior. I don't know if there is an equivalent in your SwiftFormat.

The only time I would want to wrap at the dot accessor is with builder pattern, where you have multiple dot accessors in a row (with each taking the result of the last). However in these cases the developer almost always has already wrapped at the dot accessor and we don't need SwiftFormat to do that. 

At least for my team we feel that the preferences of wrapping at the dot accessor is a dealbreaker to using the maxwidth setting.

**Comment by nicklockwood** [comment]
@1oo7 this seems to have been due to a bug in the wrapping logic, which I believe I've now fixed.

**Comment by nicklockwood** [comment]
Fixed in 0.44.8

**Comment by 1oo7** [comment]
Thank you @nicklockwood!
